### PR TITLE
v0.20.2 release for `wgpu` et al., v0.21.1 release for `wgpu-{core,hal}`

### PR DESCRIPTION
**Pull request by ErichDonGubler**
Remaining steps in the release process:

- [x] Force-push to the `v0.20` branch after this is approved.
- [x] Do a tagged release on GH.

### CODE REVIEW

File: CHANGELOG.md
Comment (cwfitzgerald) [inline CHANGELOG.md]:
```suggestion
* Do not feed `&""` to `D3DCompile`, by @workingjubilee in [#5812](https://github.com/gfx-rs/wgpu/issues/5812).
```

File: Cargo.toml
Comment (cwfitzgerald) [inline Cargo.toml]: These shouldn't get bumped - we don't need to hard depend on these.

File: Cargo.toml
Comment (cwfitzgerald) [inline Cargo.toml]: In general, unless we want to require people update a dependency, we shouldn't bump the version of the _dependency_.

File: naga/Cargo.toml
Comment (cwfitzgerald) [inline naga/Cargo.toml]: Naga didn't have any changes, so doesn't need to be bumped

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Some notes

File: Cargo.toml
Comment (ErichDonGubler) [reply Cargo.toml]: Point taken with auxiliary crates that haven't changed. `wgpu`. I reverted to previous contents on this hunk specifically. Looking good now?

But…wait, thought. Don't we _want_ to force `wgpu` users to upgrade to the fix motivating this release?I noticed wasn't updated in the 0.20.1 releaese.

File: naga/Cargo.toml
Comment (ErichDonGubler) [reply naga/Cargo.toml]: I think I've resolved this. Double-check plz? 🥺

File: CHANGELOG.md
Comment (ErichDonGubler) [reply CHANGELOG.md]: Fixed with the latest push. Looks like I need to update #5851.

**Comment by torokati44** [comment]
Do you think https://github.com/gfx-rs/wgpu/pull/5855 could/should also be backported after merging, and included in this patch release? 🤔

**Comment by ErichDonGubler** [comment]
@torokati44: If it lands first, I don't see why not. 🤷🏻‍♂️

**Comment by Wumpf** [comment]
fyi @ErichDonGubler I cherry-picked in (& resolved) a memory leak fix + changelog

File: Cargo.toml
Comment (cwfitzgerald) [inline Cargo.toml]: Missed one, but then we ship!

**Review by cwfitzgerald — APPROVED** [review]

**Comment by ErichDonGubler** [comment]
I have not yet pushed to v0.20.2 tag, but I intend to today.

NOTE: Since this PR was reviewed, <https://github.com/gfx-rs/wgpu/pull/5141> was also cherry-picked as 8a2128c9 by @nical. We got informal review on this for the `v0.20` branch specifically in our `wgpu/Naga maintainers meeting` slot today, so I don't consider this controversial.

**Comment by torokati44** [comment]
Am I correct in that the `wgpu` package on `crates.io` will stay at `v0.20.1` for this release?

**Comment by Wumpf** [comment]
correct!
~Release core core & hal still hasn't gone out yet though. Will happen soon~ Release went out by now https://github.com/gfx-rs/wgpu/releases/tag/v0.20.2

**Pull request merged**
